### PR TITLE
Add Debian 12

### LINKED ISSUE #422: Test on Debian 12
### What feature do you want to see added?

Currently, we're testing Debian 10 and 11 only when 12 is already available.

### Upstream changes

_No response_

### PR DESCRIPTION
**Pull request by NotMyFault**
Fixes #422

### CODE REVIEW

File: molecule/default/install-deb.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
     name:
       - fontconfig
       - openjdk-11-jre
```
Comment (timja) [inline molecule/default/install-deb.yml:5]: Debian 12 doesn't ship openjdk 11:
`E: Package 'openjdk-11-jre' has no installation candidate`

Either the task needs splitting with a version conditional or a different playbook file created for debian 12

**Review by basil — COMMENTED** [review]
Debian 10 only provides OpenJDK 11 and does not provide OpenJDK 17, so we would need to either remove Debian 10 or conditionally install OpenJDK 11 on Debian 10 and OpenJDK 17 on more recent Debian versions.

**Review by basil — CHANGES_REQUESTED** [review]
```Setting up ca-certificates-java (20230103) ...
Exception in thread "main" java.lang.InternalError: Error loading java.security file
	at java.base/java.security.Security.initialize(Security.java:106)
	at java.base/java.security.Security$1.run(Security.java:84)
	at java.base/java.security.Security$1.run(Security.java:82)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.base/java.security.Security.<clinit>(Security.java:82)
	at java.base/sun.security.jca.ProviderList.<init>(ProviderList.java:178)
	at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:96)
	at java.base/sun.security.jca.ProviderList$2.run(ProviderList.java:94)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.base/sun.security.jca.ProviderList.fromSecurityProperties(ProviderList.java:93)
	at java.base/sun.security.jca.Providers.<clinit>(Providers.java:55)
	at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:156)
	at java.base/java.security.cert.CertificateFactory.getInstance(CertificateFactory.java:193)
	at org.debian.security.KeyStoreHandler.<init>(KeyStoreHandler.java:50)
	at org.debian.security.UpdateCertificates.<init>(UpdateCertificates.java:65)
	at org.debian.security.UpdateCertificates.main(UpdateCertificates.java:51)
dpkg: error processing package ca-certificates-java (--configure):
 installed ca-certificates-java package post-installation script subprocess returned error exit status 1
dpkg: dependency problems prevent configuration of openjdk-17-jre-headless:amd64:
 openjdk-17-jre-headless:amd64 depends on ca-certificates-java (>= 20190405~); however:
  Package ca-certificates-java is not configured yet.

dpkg: error processing package openjdk-17-jre-headless:amd64 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of openjdk-17-jre:amd64:
 openjdk-17-jre:amd64 depends on openjdk-17-jre-headless (= 17.0.8+7-1~deb12u1); however:
  Package openjdk-17-jre-headless:amd64 is not configured yet.

dpkg: error processing package openjdk-17-jre:amd64 (--configure):
 dependency problems - leaving unconfigured
```

**Comment by timja** [comment]
https://www.reddit.com/r/debian/comments/14bypwg/comment/jokj0aq/?utm_source=reddit&utm_medium=web2x&context=3

/ https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1023748

I think

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Pull request merged**